src/tiled/automappingmanager.cpp

```diff
@@ -205,21 +206,53 @@ bool AutomappingManager::loadFile(const QString &filePath)
 void AutomappingManager::setMapDocument(MapDocument *mapDocument)
 {
     cleanUp();
```

Comment:
If you only set `mLoaded` to `false` when the rules file changes, then I don't think you can call `cleanUp` unconditionally. Probably it would be good to set `mLoaded` to `false` in `cleanUp` and to call that when the rules files changes.